src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -736,6 +737,16 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
         );
         if (target) {
             if (currentHandleIdRef.current !== OATUntargetedRelationshipName) {
+                let relationshipAmount = 0;
+                // Increment relationship amount for every item with the same type of relationship
+                elements.forEach((element) => {
+                    if (
+                        element.type === ModelTypes.relationship &&
+                        element.sourceHandle === OATRelationshipHandleName
+                    ) {
+                        relationshipAmount++;
+                    }
+                });
```

Comment:
This would be a good use case for `.reduce` (rather than `.forEach`).

There is similar logic elsewhere to calculate `relationshipAmount`. Can we reuse it?